rfc/text/wamp-terminology.md

```diff
@@ -0,0 +1,80 @@
+| *Peer*                            | A WAMP *Client* or *Router*. An implementation might embed, provide or use both roles                   |
+| *Realm*                           | Isolated WAMP URI namespace, routing and administrative domain, optionally protected by **AA**          |
+| *Transport*                       | A message-based, reliable, ordered, bidirectional (full-duplex) channel over which *Peers* communicate  |
+| *Connection*                      | An underlying entity (if any) carrying the *Transport*, e.g. a network connection, pipe, queue or such  |
+| *Session*                         | Transient conversation between a *Client* and a *Router* on a *Realm* over a *Transport*                |
+| *Message*                         | *FIXME*                                                                                                 |
+| *Serializer*                      | *FIXME*                                                                                                 |
+| *Feature*                         | *FIXME*                                                                                                 |
+| *URI (Pattern)*                   | *FIXME*                                                                                                 |
+
+
+### Authentication and Authorization (AA)
+
+{align="left"}
+| Term                               | Definition                                                                                             |
+|------------------------------------|--------------------------------------------------------------------------------------------------------|
+| *Authentication*                   | Establishes the identity of a *Session* on a *Realm*                                                   |
+| *Principal*                        | Once authenticated, *Session*s identify under a *Principal*                                            |
+| *Credentials*                      | Any authentication information and secrets used during *Authentication*                                |
+| *Authorization*                    | Decides about permission for a given *Action* on an URI or URI pattern by a *Principal*                |
+| *Access Control*                   | Policy for selective restriction of access by *Action*s on URIs or URI patterns and *Principal*s       |
+| *Role-based Access Control (RBAC)* | An *Access Control* policy based on *Realm* (`realm`), *Role* (`authrole`) of *Principal* and *Action* |
+| *Discretionary Access Control*     | *FIXME*                                                                                                |
+| *Mandatory Access Control*         | *FIXME*                                                                                                |
+| *Capability-based Access Control*  | *FIXME*                                                                                                |
```

Comment:
Well, if gonna mentioning different possible ACLs then this list is missing Attribute-based access control (ABAC)

But IDK, do we really need to describe this? I think WAMP as a protocol provides the building blocks on top of which any kind of security scheme may be built. We do not restrict or encourage or even suggest using specific security schemes. So mb it is better not to mention it at all?

Reply:
yeah, I see what you mean. on the other hand, my thinking of adding call these terms was to give the reader more clues of how stuff would fit together ... "terms systematics". fwiw, crossbar currently only implements RBAC, which is one flavor of a MAC. and I'd only be interested in - maybe - capabilities based scheme, but if so, likely only in the new router implementation I'll start ...

we should IMO have all terms that are used somewhere else in the spec text, even if only as a hint.

eg CAP is mentioned somewhere even though there is no router yet ...

anyways, I'm fine with all options: remove it for now, or I can add text coming weekend, or ...

Reply:
> we should IMO have all terms that are used somewhere else in the spec text, even if only as a hint.

Agree. That is good.